datafusion/expr/src/udaf.rs

```diff
@@ -255,15 +261,20 @@ pub trait AggregateUDFImpl: Debug + Send + Sync {
 
     /// Return a new [`Accumulator`] that aggregates values for a specific
     /// group during query execution.
-    fn accumulator(&self, arg: &DataType) -> Result<Box<dyn Accumulator>>;
+    fn accumulator(
```

Comment:
I think this API:
1. Doesn't need an owned Schema (passing in a `Schema` requires cloning which I think we can avoid
2. I am not sure why it needs an `Option` (why is it not always passed in)
3. Should have the documentation updated to explain what sort_exprs are and what schema is

Reply:
Option is because Schema is only used for ordering, so if we don't care about ordering we can just pass None.

Reply:
Or maybe we should return the schema in case we need it in the future not only for ordering? 🤔

Reply:
Unless there is some reason not to always pass in `schema` I think we should always pass it in to make for an easier to use API. For example, if it is an `Option` the user would have to call `unwrap()` or check / error when they got sort exprs